salt/cli/salt.py

```diff
@@ -425,7 +425,7 @@ def _format_ret(self, full_ret):
             if "out" in data:
                 out = data["out"]
             ret_retcode = self._get_retcode(data)
-            if ret_retcode > retcode:
+            if ret_retcode and ret_retcode > retcode:
                 retcode = ret_retcode
```

Comment:
```suggestion
retcode = ret_retcode if ret_retcode and ret_retcode > retcode else retcode
```

Alternatively, I feel like it's probably /more/ correct that instead of making the change here, we make the change in ```self._get_retcode```, which shouldn't ever be returning a non-integer value. Then this could also be rewritten as `retcode = max(retcode, ret_retcode)`, which would be a bit more readable.